broadinstitute/firecloud-orchestration
# Passthrough for searching on ORSP IDs

Addresses https://broadinstitute.atlassian.net/browse/GAWB-1759

* Passthrough for retrieving a consent from an ORSP ID.
* New models for valid consent and error cases.

Commit history is a mess, I don't recommend looking at it.

Since this touches authentication and authorization, we have a separate story to deal with FISMA changes. See https://broadinstitute.atlassian.net/browse/GAWB-1718 for addressing that aspect of this work.

Requires DUOS change so FC users (who have accounts in DUOS) can make OAuth-ed API calls:
* ~https://github.com/broadinstitute/consent/pull/233~ completed
* https://github.com/broadinstitute/firecloud-develop/pull/351/files

Requires config changes.
* Local: https://github.com/broadinstitute/firecloud-develop/pull/338
* Dev: https://github.com/broadinstitute/firecloud-develop/pull/339

## Testing Notes:
Once https://github.com/broadinstitute/firecloud-develop/pull/351 lands, you can test this branch against dev. Otherwise, spin up a local consent instance and point to `local.broadinstitute.org` in the orchestration config for consent.
* Update configs using https://github.com/broadinstitute/firecloud-develop/pull/338
* Spin up local orchestration
* Sign into swagger as a user that exists in both FC and DUOS.
* Query on some ORSP IDs. Good ones include:
  * OD-310 (should be not found)
  * OD-320 (should be valid)
  * OD-320 (should be not approved) 
  * Any random string
* Also log in as a user in FC, but not in DUOS. Any query should get a 401.

---

  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
